### PR TITLE
Added Select

### PR DESCRIPTION
**Pull request by alansouzati**
Also refactored a bunch of other things.

select

### CODE REVIEW

File: src/js/components/Select/doc.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      'An select-like field with optional search capability.'
+    ).usage(
+      `import { Select } from 'grommet';
+      <Select/>`
+    );
+
+  DocumentedSelect.propTypes = {
+    a11yTitle: a11yTitlePropType,
+    background: backgroundPropType,
+    basis: PropTypes.string.description(
```
Comment (ericsoderberghp) [inline src/js/components/Select/doc.js]: `basis` seems like an odd name for this property. I understand how it is being used, but the caller isn't aware of the context. Can we not automatically handle this without having the caller specify it? Could we could do something like `flex: 0 1 80vh` all the time?

File: src/js/components/Select/doc.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      'Size of the options container inside the Select drop.'
+    ),
+    children: PropTypes.func.description(
+      'Function that will be called when each option is rendered.'
+    ),
+    name: PropTypes.string.description(
```
Comment (ericsoderberghp) [inline src/js/components/Select/doc.js]: Can we just include this in a `...rest` and not need to document it?

File: src/js/components/Select/doc.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      'Whether the Select drop should be open or not.'
+    ),
+    options: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.object])
+    ).description(
+      `Options can be either a string or an object. If an object is used, use children callback
+      in order to render anything based on the current item.`
+    ).isRequired,
+    placeholder: PropTypes.string.description(
+      'Placeholder text to use when the input is empty.'
```
Comment (ericsoderberghp) [inline src/js/components/Select/doc.js]: 'when the input is empty' -> 'when no value is provided'

File: src/js/components/Select/doc.js (hunk 1 above)
Comment (alansouzati) [reply src/js/components/Select/doc.js]: let me try and see if it works. great suggestion!

File: src/js/components/Select/doc.js (hunk 2 above)
Comment (alansouzati) [reply src/js/components/Select/doc.js]: we can definitely do this. but I'm using the name inside select

https://github.com/grommet/grommet/pull/1686/files#diff-e73ec96fafca62459d0898bace1bd91fR151

File: src/js/components/Select/doc.js (hunk 3 above)
Comment (alansouzati) [reply src/js/components/Select/doc.js]: good

**Pull request merged**